xla/service/cpu/tests/onednn_convolution_test.cc

```diff
@@ -32,159 +33,221 @@ limitations under the License.
-    ROOT add.10 = f32[1,11,11,1] add(convolution.0, broadcast.9)
+TEST_P(ConvolutionTest, Conv2DWithBinaryAddTest) {
+  const absl::string_view outline = R"(
+  HloModule convolution.test.with.binaryadd
+
+  ENTRY convolution.test.with.binaryadd {
+    arg0.1 = $0[1,22,22,1] parameter(0)
+    constant.3 = $0[] constant(1)
+    broadcast.4 = $0[8,8,1,1] broadcast(constant.3), dimensions={}
+    convolution.0 = $0[1,11,11,1] convolution(arg0.1, broadcast.4), window={size=8x8 stride=2x2 pad=3_3x3_3}, dim_labels=b01f_01io->b01f
+    constant.5 = $0[] constant(15)
+    broadcast.6 = $0[1] broadcast(constant.5), dimensions={}
+    broadcast.9 = $0[1,11,11,1] broadcast(broadcast.6), dimensions={3}
+    ROOT add.10 = $0[1,11,11,1] add(convolution.0, broadcast.9)
   })";
 
-  EXPECT_TRUE(RunAndCompare(convolution_module_str, ErrorSpec{1e-4, 1e-4}));
-  MatchOptimizedHlo(convolution_module_str, fused_convolution_binary_add_);
+  RunCompareAndMatchOptimizedHlo(outline, {"BINARY_ADD"});
 }
 
-// This test should match BIAS + Residual Add when the residual add fusion is
+// This test should match BIAS + RESIDUAL ADD when the residual add fusion is
 // re-enabled.
-TEST_F(ConvolutionTest, SimpleTestBF16WithBiasAndAddFusion) {
-  const char* convolution_module_str = R"(
-  HloModule convolution.add.test.bf16
-
-  ENTRY convolution.add.test.bf16 {
-    arg0.1 = bf16[1,22,22,1] parameter(0)
-    arg0.2 = bf16[8,8,1,10] parameter(1)
-    convolution.0 = bf16[1,11,11,10] convolution(arg0.1, arg0.2), window={size=8x8 stride=2x2 pad=3_3x3_3}, dim_labels=b01f_01io->b01f
-    const.0 = bf16[10] constant(15)
-    bcast.1 = bf16[1,11,11,10] broadcast(const.0), dimensions={3}
-    add.0 = bf16[1,11,11,10] add(convolution.0, bcast.1)
-    const.1 = bf16[1,11,11,10] constant({...})
-    ROOT add.1 = bf16[1,11,11,10] add(add.0, const.1)
+TEST_P(ConvolutionTest, Conv2DWithBiasAndBinaryAddTest) {
+  const absl::string_view outline = R"(
+  HloModule convolution.add.test
+
+  ENTRY convolution.add.test {
+    arg0.1 = $0[1,22,22,1] parameter(0)
+    arg0.2 = $0[8,8,1,10] parameter(1)
+    convolution.0 = $0[1,11,11,10] convolution(arg0.1, arg0.2), window={size=8x8 stride=2x2 pad=3_3x3_3}, dim_labels=b01f_01io->b01f
+    const.0 = $0[10] constant(15)
+    bcast.1 = $0[1,11,11,10] broadcast(const.0), dimensions={3}
+    add.0 = $0[1,11,11,10] add(convolution.0, bcast.1)
+    const.1 = $0[1,11,11,10] constant({...})
+    ROOT add.1 = $0[1,11,11,10] add(add.0, const.1)
   })";
 
-  EXPECT_TRUE(RunAndCompare(convolution_module_str, ErrorSpec{1e-2, 1e-2}));
-  MatchOptimizedHlo(convolution_module_str, conv_rewrite_bias_str_);
+  RunCompareAndMatchOptimizedHlo(outline, {"BIAS"});
 }
 
+INSTANTIATE_TEST_SUITE_P(
+    OneDnnConvolutionTestSuite, ConvolutionTest,
+    ::testing::Values(F32, BF16, F16),
+    [](const ::testing::TestParamInfo<ConvolutionTest::ParamType>& info) {
+      auto infoString = primitive_util::LowercasePrimitiveTypeName(info.param);
```

Comment:
Nit: Let's rename the returned variable to `test_name` to make it more obvious.
```suggestion
      auto test_name = primitive_util::LowercasePrimitiveTypeName(info.param);
```

Reply:
Done